src/test/kotlin/uk/gov/justice/digital/hmpps/locationsinsideprison/resource/LocationResourceIntTest.kt

```diff
@@ -734,6 +735,115 @@ class LocationResourceIntTest : SqsIntegrationTestBase() {
+
+      @Test
+      fun `duplicate location is rejected`() {
+        webTestClient.post().uri("/locations/create-wing")
+          .headers(setAuthorisation(roles = listOf("ROLE_MAINTAIN_LOCATIONS"), scopes = listOf("write")))
+          .header("Content-Type", "application/json")
+          .bodyValue(jsonString(createWingRequest.copy(wingCode = "Z")))
+          .exchange()
+          .expectStatus().is4xxClientError
+      }
+    }
+
+    @Nested
+    inner class HappyPath {
+      @Test
+      fun `can create an entire wing with 3 landings, 2 spurs and 20 cells per spur`() {
+        webTestClient.post().uri("/locations/create-wing")
+          .headers(setAuthorisation(roles = listOf("ROLE_MAINTAIN_LOCATIONS"), scopes = listOf("write")))
+          .header("Content-Type", "application/json")
+          .bodyValue(createWingRequest)
+          .exchange()
+          .expectStatus().isCreated
+          .expectBody().json(
+            // language=json
+            """ 
+             {
+              "prisonId": "MDI",
+              "code": "Y",
+              "pathHierarchy": "Y",
+              "locationType": "WING",
+              "active": true,
+              "key": "MDI-Y",
+              "description": "Y Wing",
+              "orderWithinParentLocation": 1,
+              "capacity": {
+                "capacity": 120,
+                "operationalCapacity": 120
+              },
+              "certification": {
+                "certified": true,
+                "capacityOfCertifiedCell": 120
+              }
+            }
+          """,
+            false,
+          )
+      }
+
+      @Test
+      fun `can create an entire wing with 4 landings, 20 cells per landing`() {
+        webTestClient.post().uri("/locations/create-wing")
+          .headers(setAuthorisation(roles = listOf("ROLE_MAINTAIN_LOCATIONS"), scopes = listOf("write")))
+          .header("Content-Type", "application/json")
+          .bodyValue(createWingRequest.copy(wingCode = "X", wingDescription = "X Wing", numberOfLandings = 4, numberOfSpursPerLanding = 0, numberOfCellsPerSection = 20, defaultCellCapacity = 2, defaultAttributesOfCells = setOf(ResidentialAttributeValue.DOUBLE_OCCUPANCY)))
+          .exchange()
+          .expectStatus().isCreated
+          .expectBody().json(
+            // language=json
+            """ 
+             {
+              "prisonId": "MDI",
+              "code": "X",
+              "pathHierarchy": "X",
+              "locationType": "WING",
+              "active": true,
+              "key": "MDI-X",
+              "description": "X Wing",
+              "orderWithinParentLocation": 1,
+              "capacity": {
+                "capacity": 160,
+                "operationalCapacity": 160
+              },
+              "certification": {
+                "certified": true,
+                "capacityOfCertifiedCell": 160
+              }
+            }
+          """,
+            false,
+          )
+      }
+    }
```

Comment:
I was going to ask if you were intending to read back the created data to verify it, but presumably the endpoint is basically to aid testing rather than being required in prod?

Reply:
I've just read back the capacities to know that all the cells below have been aggregated up.  Doubt we'd use it in prod, but front end might support the "creation" of a new wing